### ISSUE TITLE
bug: fails to compile on win11 due to 'static_assert' is undefined

### ISSUE THREAD

**Issue by Parad0x84** [tree-sitter/tree-sitter-cpp#275]
### Did you check existing issues?

- [X] I have read all the [tree-sitter docs](https://tree-sitter.github.io/tree-sitter/using-parsers) if it relates to using the parser
- [X] I have searched the existing issues of tree-sitter-cpp

### Describe the bug

treesitter fails to compile tree-sitter-cpp

output log: (note: I intentionally didn't change formatting of log message, since I don't know what's what)
```
nvim-treesitter[cpp]: Error during compilation
src/scanner.c: In function 'tree_sitter_cpp_external_scanner_serialize':^M                                           src/scanner.c:126:5: warning: implicit declaration of function 'static_assert' [-Wimplicit-function-declaration]^M        static_assert(MAX_DELIMITER_LENGTH * sizeof(wchar_t) < TREE_SITTER_SERIALIZATION_BUFFER_SIZE,^M                      ^~~~~~~~~~~~~^M                                                                                                 C:\Users\admin\AppData\Local\Temp\ccKywmf4.o:scanner.c:(.text+0x199): undefined reference to `static_assert'^M       collect2.exe: error: ld returned 1 exit status^M 
```

### Steps To Reproduce/Bad Parse Tree

I simply added tree-sitter and put 'cpp' to the ensure_installed field
Note: OS is Win11

### Expected Behavior/Parse Tree

I expected it to simply work

**Comment by LightningPheonix77** [comment]
It also happened to me on win10 mingw gcc compiler

**Comment by amaanq** [comment]
Ensure you're compiling with the std set to C11, as that's the minimum required version for tree-sitter.

**Comment by xflcx1991** [comment]
> Ensure you're compiling with the std set to C11, as that's the minimum required version for tree-sitter.

@amaanq  But how to specify that the std standard is C11, with inspect.signature(Language.build_library) there are only 2 parameters: output_path and repo_paths.

**Comment by peterzsj6** [comment]
I had same issue with linux:
gcc --version
g++ --version
gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
g++ (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
tree_sitter              0.20.1

Error message I got:
tree-sitter-languages/tree-sitter-cpp/src/scanner.c:126:5: warning: implicit declaration of function ‘static_assert’ [-Wimplicit-function-declaration]
  126 |     static_assert(MAX_DELIMITER_LENGTH * sizeof(wchar_t) < TREE_SITTER_SERIALIZATION_BUFFER_SIZE,
      |     ^~~~~~~~~~~~~


while loading language.so:
OSError: /data/home/zhangsj/lora_code/ast_parsing_folder/my-languages.so: undefined symbol: static_assert



> Ensure you're compiling with the std set to C11, as that's the minimum required version for tree-sitter.

**Comment by ObserverOfTime** [comment]
tree-sitter-languages is dead. Stop using it.

**Comment by Feng-Jay** [comment]
Hi! @peterzsj6 
Do you have a workaround for this issue? I think one way is to replace `static_assert` with `_Static_assert`, but I don't think that is an elegant solution.

**Comment by Feng-Jay** [comment]
I think I find a workaround

TL;DR

If you are using the deprecated API `Language.build_library`, and have upgraded to GCC 11, you should also **use version 0.21.3 of py-tree-sitter** to ensure compatibility.

Here is a detailed analysis

This is the example code that you use `tree-sitter-cpp` to build your own `.so`
```python
# build.py
from tree_sitter import Language, Node, Parser

CURR = Path(__file__).resolve().parent.absolute()

Language.build_library(
    str(CURR / "build/my-languages.so"),
    [str(CURR / "tree-sitter-cpp")],
)
```

We can print the full stack trace on Ubuntu to identify GCC-related commands using:

```bash
strace -f -s 4096 -e trace=execve python build.py 2>&1 | grep gcc
```
When using py-tree-sitter v0.20.4 or other incompatible versions, the output shows:
```bash
[pid 1486397] execve("/usr/libexec/gcc/x86_64-linux-gnu/13/cc1", ["/usr/libexec/gcc/x86_64-linux-gnu/13/cc1", "-quiet", "-I", ".../vendor/tree-sitter-cpp/src", "-imultiarch", "x86_64-linux-gnu", ".../vendor/tree-sitter-cpp/src/scanner.c", ...**"-std=c99",** "-fPIC", ...
```
This indicates that -std=c99 is hardcoded and cannot be overridden by changing the default GCC version or environment variables.

Upgrading to py-tree-sitter v0.21.3 resolves this issue, as the build now uses:
```bash
[pid 1487958] execve("/usr/libexec/gcc/x86_64-linux-gnu/13/cc1", ["/usr/libexec/gcc/x86_64-linux-gnu/13/cc1", "-quiet", "-I", ".../vendor/tree-sitter-cpp/src", "-imultiarch", "x86_64-linux-gnu", ".../vendor/tree-sitter-cpp/src/scanner.c", ..., **"-std=c11",** "-fPIC",...

```
This change ensures compatibility with newer GCC versions.

Since the Language.build_library API is no longer supported in the official repository, I wouldn’t consider this a bug. However, for those who still need to use this API with `tree-sitter-cpp`, the solution above is helpful—and arguably a much more elegant workaround.
